src/naive/datetime/mod.rs

```diff
@@ -728,28 +728,20 @@ impl NaiveDateTime {
     ///
     /// # Errors
     ///
-    /// Returns `None` if:
-    /// - The resulting date does not exist (February 29 in a non-leap year).
-    /// - The year is out of range for a `NaiveDate`.
+    /// - See [`NaiveDate::with_year`] if the intermediate conversion to [`NaiveDate`] fails.
+    /// - Cannot fail on the [`NaiveTime`] component.
```

Comment:
I prefer to just repeat the error cases instead of letting users chaise links in the documentation. It is not going to increase the number of lines :smile:.

Reply:
The next level is `DateTime`, and that only introduces one more case through the `LocalResult` type (by its new name by then). And that will be encoded in the method signature. I'm not afraid yet.

When it comes to the parsing methods there is an argument for linking to the documentation of `Parsed`. Let's see if it gets unwieldy.